spike_main/spike.cc

```diff
@@ -117,27 +117,24 @@ static void read_file_bytes(const char *filename,size_t fileoff,
 
 bool sort_mem_region(const mem_cfg_t &a, const mem_cfg_t &b)
 {
-  if (a.base == b.base)
-    return (a.size < b.size);
+  if (a.get_base() == b.get_base())
+    return (a.get_inclusive_end() < b.get_inclusive_end());
```

Comment:
Is this a functional change if one of these memories wraps around the 64-bit address space? Or have we already forbidden such memories before we get to this code?

Reply:
this was not a functional change (since we've already forbidden such regions because of assertions). However I've updated the PR and left this place without significant changes (except for the usage of getters)